src/large_pages/node_large_page.cc

```diff
@@ -373,13 +371,16 @@ MoveTextRegionToLargePages(const text_region& r) {
     return -1;
   }
 #elif defined(__APPLE__)
+  // There is not enough room to reserve the mapping close
+  // to the region address so we content to give a hint
+  // without forcing the new address being closed to.
+  // We explicitally gives all permission since we plan
   tmem = mmap(start, size,
               PROT_READ | PROT_WRITE | PROT_EXEC,
               MAP_PRIVATE | MAP_ANONYMOUS,
               VM_FLAGS_SUPERPAGE_SIZE_2MB, 0);
   if (tmem == MAP_FAILED) {
     PrintSystemError(errno);
-    munmap(nmem, size);
```

Comment:
There are a few more `munmap(nmem, size)` calls in linux/freebsd code paths. Can you remove those as well?